service/forward_service.cc

```diff
@@ -412,27 +398,50 @@ future<query::forward_result> forward_service::execute_on_this_shard(
         cql3::query_options::specific_options::DEFAULT,
         cql_serialization_format::latest()
     );
-    auto pager = service::pager::query_pagers::pager(
-        _proxy,
-        schema,
-        selection,
-        *query_state,
-        *query_options,
-        make_lw_shared<query::read_command>(std::move(req.cmd)),
-        retain_ranges_owned_by_this_shard(schema, std::move(req.pr)),
-        nullptr // No filtering restrictions
-    );
+
     auto rs_builder = cql3::selection::result_set_builder(
         *selection,
         now,
         cql_serialization_format::latest(),
         std::vector<size_t>() // Represents empty GROUP BY indices.
     );
 
-    // Execute query.
-    while (!pager->is_exhausted()) {
-        co_await pager->fetch_page(rs_builder, DEFAULT_INTERNAL_PAGING_SIZE, now, timeout);
-    }
+    dht::partition_range_vector ranges_owned_by_this_shard;
+    partition_ranges_owned_by_this_shard owned_iter(schema, std::move(req.pr));
+
+    std::optional<dht::partition_range> current_range;
+    do {
+        // We serve up to 256 ranges at a time to avoid allocating a huge vector for ranges
+        static constexpr size_t max_ranges = 256;
```

Comment:
256 means it's never really tested.

Reply:
What do you mean? I tested a local table and it used to produce 4096 or 4097 ranges at a time before the patch, and the same in 256-sized batches after the patch

Reply:
I assumed the previous thing meant that we're only seeing one row (due to 

```c++
        if (rows.size() != 1) {
            flogger.error("aggregation result row count != 1");
            throw std::runtime_error("aggregation result row count != 1");
        }
```
)

and therefore that the ranges got squished. But apparently if builder self-aggregates, that doesn't happen,